File: hw5/circle.cpp
#include "circle.h"
#include "shape.h"
using namespace std;
using namespace Shape_;


//------------------------------------CONSTRUCTORS AND OTHER FUNCS-------------------------------------

 //no parameter constructor
Circle::Circle(): radius(0.0), coorX(0.0), coorY(0.0){}

Circle::Circle(int coorValue1, int coorValue2)
{
    coorX = coorValue1;
    coorY = coorValue2;
}

Circle::Circle(double radiusValue)
{
    radius = radiusValue;
}

//draw function for container shape
void Circle::draw(ostream &fp)
{
    fp << "<circle cx=\" " << coorX << "\" cy=\" " << coorY << "\" r=\" " << radius <<
       "\" fill=\"green\" stroke=\"black\" stroke-width=\"0.3\" />" << endl;
}

//calculates container's perimeter length
double Circle::perimeter()
{
    double objPer = 2 * 3.14 * radius;
    shapePer = objPer;
    return shapePer;
}

//calculates container's area
double Circle::area()
{
    double objArea = 3.14 * radius * radius;
    shapeArea = objArea;
    return shapeArea;

}

//--------------------------------------OP. OVERLOADING FUNCTIONS------------------------------------------------------

//overloads << to draw container
ostream& operator <<(ostream& fp, const Circle& object)
{
    fp << "<circle cx=\" " << object.getCoorX() << "\" cy=\" " << object.getCoorY() << "\" r=\" " << object.getRadius() <<
       "\" fill=\"red\" stroke=\"black\" stroke-width=\"0.3\" />" << endl;
    return fp;
}

//overload ++ to increment coordinate -> prefix version
Shape& Circle::operator++()
{
    Shape* ptr = new Circle(coorX++, coorY++);
    return *ptr;
}

//overloads ++ to increment coordinate -> postfix version
Shape& Circle::operator++(int x)
{
    double temp1= coorX;
    double temp2 = coorY;
    coorX ++;
    coorY ++;
    Shape* ptr = new Circle(temp1, temp2);
    return *ptr;
}

//overload -- to increment coordinate -> prefix version
Shape&  Circle::operator--()
{
    Shape* ptr = new Circle(coorX--, coorY--);
    return *ptr;
}

//overloads -- to increment coordinate -> postfix version
Shape&  Circle::operator--(int x)
{
    double temp1 = coorX;
    double temp2 = coorY;
    coorX--;
    coorY--;
    Shape* ptr = new Circle(temp1, temp2);
    return *ptr;
}

//overloads + to increment length and return a new object
const Circle Circle::operator +(double value)
{
    double newRadius = radius + value;
    return Circle(newRadius);
}
//overloads - to decrement length and return a new object
const Circle Circle::operator -(double value)
{
    double newRadius = radius - value;
    return Circle(newRadius);
}

//overloads == to compare shapes respect to their areas
/*bool Circle::operator ==(const Circle& obj) const
{
    return (area == obj.area);
}

//overloads != to compare shapes respect to their areas
bool Circle::operator !=(const Circle& obj) const
{
    return (area != obj.area);
}*/

//-------------------------------GETTERS---------------------------------------------------------------

double Circle::getRadius() const
{   return radius;
}

double Circle::getCoorX() const
{   return coorX;
}

double Circle::getCoorY() const
{   return coorY;
}



//------------------------------SETTERS------------------------------------------------------------------

void Circle::setRadius(double setRadius)
{   radius = setRadius;
}

void Circle::setCoorX(double setCoorX)
{   coorX = setCoorX;
}

void Circle::setCoorY(double setCoorY)
{   coorY = setCoorY;
}

File: hw7/Circle.java
import java.awt.Color;
import java.awt.Graphics;
/**
 * 
 * @author GONCA EZGI CAKIR
 *
 */
public class Circle implements Shape
{
	@SuppressWarnings("unused")
	private static final String Graphic2D = null;
	//radius of circle
	private double radius;
	//coorX of circle's center
	private double coorX; 
	//coorY of circle's center  
	private double coorY; 
	private boolean flag;
	
	/**
	 * constructor
	 */
	public Circle()
	{
		radius = 0.0;
		coorX = 0.0;
		coorY = 0.0;
		flag = false;
	}

	/**
	 * constructor
	 * @param coorValue1 int
	 * @param coorValue2 int
	 */
	public Circle(int coorValue1, int coorValue2)
	{
	    coorX = coorValue1;
	    coorY = coorValue2;
	    flag = false;
	}

	/**
	 * constructor
	 * @param radiusValue double
	 */
	public Circle(double radiusValue)
	{
	    radius = radiusValue;
	    flag = false;
	}
	

	
	//-------------------------------GETTERS---------------------------------------------------------------
	
	
	/**
	 * getter for radius
	 * @return double
	 */
	public double getRadius()
	{   return radius;
	}
	
	/**
	 * getter for coorX
	 * @return double
	 */
	public double getCoorX()
	{   return coorX;
	}
	
	/**
	 * getter for coorY
	 * @return double
	 */
	public double getCoorY()
	{   return coorY;
	}

	//-------------------------------SETTERS---------------------------------------------------------------
	/**
	 * setter for radius
	 * @param setRadius double
	 */
	public void setRadius(double setRadius)
	{   radius = setRadius;
	}
	
	/**
	 * setter for coorX
	 * @param setCoorX double
	 */
	public void setCoorX(double setCoorX)
	{   coorX = setCoorX;
	}
	
	/**
	 * setter coorY
	 * @param setCoorY double
	 */
	public void setCoorY(double setCoorY)
	{   coorY = setCoorY;
	}
	
	/**
	 * setter for flag
	 * @param val boolean
	 */
	public void setFlag(boolean val)
	{
		flag = val;
	}
	

	//---------------------------------------------------------------------------------------------------
	@Override
	/**
	 * area function
	 * @return double
	 */

	public double area() 
	{
		 double objArea = 3.14 * radius * radius;
		 return objArea;	
	}

	@Override
	/**
	 * perimeter function
	 * @return double
	 */
	public double perimeter() 
	{
	    double objPer = 2 * 3.14 * radius;
	    return objPer;	
	}

	@Override
	/**
	 * draw function
	 * @param obj Graphics refrence
	 */

	public void draw(Graphics obj){
		if(flag)
		{
			obj.fillOval((int)(coorX-radius), (int)(coorY-radius), (int)(radius*2), (int)(radius*2));
			obj.setColor(Color.BLACK);
			obj.fillOval((int)(coorX-radius), (int)(coorY-radius), (int)(radius*2), (int)(radius*2));
		}
		else
		{
			obj.fillOval((int)(coorX), (int)(coorY), (int)(radius*2), (int)(radius*2));
			obj.setColor(Color.BLACK);
			obj.drawOval((int)(coorX), (int)(coorY), (int)(radius*2), (int)(radius*2));
		}
		
	}

	@Override
	/**
	 * increment function
	 * @param obj Graphics refrence
	 */
	public void increment(Shape obj){}

	@Override
	/**
	 * decrement function
	 * @param obj Graphics refrence
	 */
	public void decrement(Shape obj){}
	
	
	
}
